app/events/applications/application_created_event.rb

```diff
@@ -3,6 +3,7 @@ class Applications::ApplicationCreatedEvent < ApplicationRelatedEvent
   # @param [User] user
   def self.create(application, user)
     provider = application.provider_account
+    service = application.service || Service.new({id: application.service_id}, without_protection: true)
```

Comment:
why would `application.service` be `nil`?

Reply:
It's fine, I guess you copied from `Applications::ApplicationDeletedEvent` but still an interesting question.

Reply:
> I guess you copied from `Applications::ApplicationDeletedEvent` but still an interesting question.

You're right :sweat_smile: 
And indeed an interesting question... In fact, we might not need this here. Probably, in the "delete" scenario it was just to prevent failures when an application is being deleted, but the service already doesn't exist, because it was deleted before... 
But I am not sure in fact whether this scenario is possible... I'll investigate.

Reply:
I'm also trying to understand this warning https://github.com/3scale/porta/blob/master/app/events/applications/application_created_event.rb#L11-L13

but can't figure out the problem so far...

Reply:
me too, even more because underneath I see `service_id: application.service_id`. Maybe need to change and run the suggested spec. But also not sure it is important to spend time on it now.

Reply:
if tests pass, I guess it is outdated